cmd/enterprise-portal/internal/dotcomdb/dotcomdb.go

```diff
@@ -453,3 +459,37 @@ func (r *Reader) ListEnterpriseSubscriptionLicenses(
 	}
 	return attrs, rows.Err()
 }
+
+type SubscriptionAttributes struct {
+	ID         string
+	CreatedAt  time.Time
+	ArchivedAt *time.Time
+}
+
+// ListEnterpriseSubscriptions returns a list of enterprise subscription
+// attributes with the given IDs. It silently ignores any non-existent
+// subscription IDs. The caller should check the length of the returned slice to
+// ensure all requested subscriptions were found.
+func (r *Reader) ListEnterpriseSubscriptions(ctx context.Context, subscriptionIDs ...string) ([]*SubscriptionAttributes, error) {
+	if len(subscriptionIDs) == 0 {
+		return []*SubscriptionAttributes{}, nil
+	}
+
+	idConds := sqlf.Join(dbutil.ArgsToQueries(subscriptionIDs), ", ")
+	q := sqlf.Sprintf("SELECT id, created_at, archived_at FROM subscriptions WHERE id IN (%s)", idConds)
+	rows, err := r.db.Query(ctx, q.Query(sqlf.PostgresBindVar), q.Args()...)
```

Comment:
85% sure you can do:

```suggestion
	rows, err := r.db.Query(ctx, "SELECT id, created_at, archived_at FROM subscriptions WHERE id ANY ($1)", subscriptionIDs)
```

Reply:
or a named arg if you'd like :-)